geps/gep-1619.md

```diff
@@ -62,14 +62,110 @@ Header-based stateful sessions are achieved by a backend or gateway providing an
 
 Session information can be also encoded into the request URL to establish a persistent session. The server rewrites the client's URL to encode the new session information automatically. The server then decodes the session information from the URL to identify the session.
 
-**Session Initiation**
+### Session Persistence Initiation
 
-For both header-based and cookie-based sessions, either the gateway or the backend server can initiate establishing the session via the appropriate header or set-cookie attributes. The following rules apply based on who initiates the session:
+For both header-based and cookie-based sessions, the initiation of session persistence is possible from various sources,
+including the gateway, backend, a sidecar in a backend, or any other infrastructure component. The client application
+also plays in a role in session initiation. If a client doesn't want a request to belong an existing session, it can
+remove the session cookie or identifier to signal the gateway or backend to recreate the session.
 
+Let's focus on the two primary sources: gateways and the backend. The following rules apply based on who initiates the session:
+- If the gateway initiates the session, the backend will be presented with session attributes regardless if it enabled them.
```

Comment:
Given the next statement, do you mean the API should have 2 modes, one where Gateway force-adds affinity based on a setting, and the other where the Gateway defers to backend to start the session (to support Servlet/PHP/etc APIs) ?

I think it would be nice - but may not be easy to implement with envoy ( worth doing it eventually...)

May worth adding a line justifying the 'not force persistence but defer to bakend' use case for "languages/framework that define API allowing backend to initiate (or disable) the affinity, like Servlet/PHP"

Reply:
> If the gateway initiates the session, the backend will be presented with session attributes regardless if it enabled them
Not sure how such session attributes would be presented to the backend ? Did you perhaps mean on future requests where the client would be providing a cookie value that was previously set by the gateway? or is it something else ?